lpd/models.py

```diff
@@ -22,3 +26,219 @@ class LearnerProfileDashboardForm(forms.ModelForm):
+        choices=QUESTION_TYPES,
+        max_length=20,
+        help_text='Whether this question requires learners to produce a short answer or an essay.',
+    )
+    influences_group_membership = models.BooleanField(
+        help_text=(
+            'Whether answers to this question should be taken into account '
+            'when calculating group membership for specific learners.'
+        ),
+    )
+
+
+class QuantitativeQuestion(Question):
+    """
+    Abstract base class for models representing questions with a pre-defined set of answer options.
+    """
+    randomize_options = models.BooleanField(
+        help_text='Whether to display answer options in random order on LPD.',
+    )
+
+    class Meta(OrderedModel.Meta):
+        abstract = True
+
+    def get_content_type(self):
+        return ContentType.objects.get_for_model(self).id
+
+
+class MultipleChoiceQuestion(QuantitativeQuestion):
+    """
+    Represents a multiple choice question (MCQ) or multiple response question (MRQ).
+    """
+    max_options_to_select = models.PositiveIntegerField(
+        help_text=(
+            'Maximum number of answer options that learner is allowed to select for this question. '
+            'Set this to 1 to create a multiple choice question. '
+            'Set this to a value larger than one to create a multiple response question.'
+        ),
+    )
+
+
+class RankingQuestion(QuantitativeQuestion):
+    """
+    Represents a question that asks learners to rank (a subset of) its answer options,
+    on a scale from 1 to `number_of_options_to_rank`.
+    """
+    number_of_options_to_rank = models.PositiveIntegerField(
+        help_text='Number of answer options belonging to this question that learners are allowed to rank.',
+    )
+
+
+class LikertScaleQuestion(QuantitativeQuestion):
+    """
+    Represents a (simplified) Likert Scale question, cf. https://en.wikipedia.org/wiki/Likert_scale.
+    """
+    answer_option_range = models.PositiveIntegerField(
+        help_text=(
+            'Number of values that learners can choose from for each answer option. '
+            'For example, to create 5-point scale, set this to 5.'
+        ),
+    )
+    range_min_text = models.CharField(
+        max_length=50,
+        help_text='Meaning of lowest value of Likert scale. For example: "Not very valuable".',
+    )
+    range_max_text = models.CharField(
+        max_length=50,
+        help_text='Meaning of highest value of Likert scale. For example: "Extremely valuable."',
+    )
+
+
+class AnswerOption(OrderedModel):
+    """
+    Represents a specific answer option for a quantitative learner profile question.
+    """
+    # Use generic relation to connect this model to QuantitativeQuestion (which is abstract).
+    question_type = models.ForeignKey(ContentType)
+    question_id = models.PositiveIntegerField()
+    question_object = GenericForeignKey('question_type', 'question_id')
+
+    knowledge_component = models.OneToOneField(
```

Comment:
Question: shouldn't this be actually inside question models? Or every answer option in the question might be related to the different knowledge component?

Reply:
@tomaszgy 

> Question: shouldn't this be actually inside question models? Or every answer option in the question might be related to the different knowledge component?

That's right: It's not *questions* that have knowledge components (tags) associated with them, but rather individual answer options :)

The TLL team wants the adaptive engine to tune recommendations based on a learner's interests. So the LPD might include a question like:

> What type of job would you like to work in after this course?

And answers might include options such as:

* Teacher
* Firefighter
* Biologist

The LPD will send a score for each one of these options to the adaptive engine, so that the adaptive engine can recommend different types of content based on the option that a learner selected.

---

This was a bit confusing to me at first as well, but I find that having a concrete example in the back of my mind really helped :)

Reply:
Yeah, a concrete example helped a lot, thanks! ;)